### DISCUSSION TITLE
Review Pull Request Step 1 Issue

### DISCUSSION THREAD

**Discussion by reeba212** [skills/.github#157] [Q&A]
I want to complete the review-pull-request module. The first step is to open a pull request and assign ourselves as the reviewer. However, I am not able to see any pull requests even after refreshing several times after several minutes. 
<img width="960" alt="git1" src="https://user-images.githubusercontent.com/64488642/215412498-2deedc9d-1e95-431f-aebd-10ed17fd76c8.png">
After trying to figure it out I came upon the actions page which showed that there is some error and GitHub is not able to create the pull request. 
<img width="960" alt="git2" src="https://user-images.githubusercontent.com/64488642/215412528-200fdf0c-1a4b-4e13-adf3-26e7b925378f.png">
I updated my settings and allowed GitHub to create pull requests.
<img width="960" alt="git3" src="https://user-images.githubusercontent.com/64488642/215412541-ba7933f1-0e72-4a74-869b-f1b0b7804a00.png">
However, the error remains and I am not able to see any pull requests.
Can you please help me out and tell me what to do next?

**Comment by sinsukehlab** [comment]
The action actions/checkout used in the actions is old.
I will bump it to v3 like skills/communicate-using-markdown#22.

**Reply by sinsukehlab** [reply]
Oh, that is just the reason of the warning.

**Comment by sinsukehlab** [comment]
The error occurred because of the restricted permission. The workflow runs when the initial commit is pushed to main and the permission at that time is restricted. This restriction can probably be circumvented by running shell scripts instead of running commands.

**Reply by sinsukehlab** [reply]
I wonder how the maintainers deal with the restriction. They are planning to change the courses to let learners create PRs. This strategy works for most of the courses, but you can't approve your own PR.

**Reply by sinsukehlab** [reply]
Oh, the restriction cannot be circumvented by running shell scripts. The initial workflow in the release-based-workflow course doesn't exit with error, but it fails in creating a PR.

**Reply by sinsukehlab** [reply]
Now that the course doesn't use `gh pr create`, the error doesn't occur.
See: skills/review-pull-requests#29

**Reply by sinsukehlab** [reply]
I have confirmed that the error doesn't occur now.
